bindings/java/fdbJNI.cpp

```diff
@@ -1042,8 +1036,43 @@ JNIEXPORT void JNICALL Java_com_apple_foundationdb_FDB_Network_1stop(JNIEnv *jen
 }
 
 jint JNI_OnLoad(JavaVM *vm, void *reserved) {
+    JNIEnv *env;
 	g_jvm = vm;
-	return JNI_VERSION_1_1;
+	if (vm->GetEnv((void**)&env, JNI_VERSION_1_6) != JNI_OK) {
```

Comment:
What is the significance of this version being different than the one we return (1.6 vs. 1.1)?

Reply:
I believe we are expecting at least 1.6 features in some of our existing calls.  Let me get more precise here on which function calls go with which version of JNI.  Thanks for reviewing appreciate it.

Reply:
I was wrong...  We are at 1_4 I believe in the C Code without callbacks into java.  The code however calls into java that would require at least 1.5 (generics).

https://docs.oracle.com/javase/8/docs/technotes/guides/jni/jni-14.html. 

We utilize

```
AttachCurrentThreadAsDaemon
```

How would we feel requiring at least JDK 1.6?  @ajbeamon

Reply:
I think our bindings require a minimum Java version of 8 (or later?), so that sounds ok to me.

Reply:
Does this require a change to return a new version from JNI_OnLoad?

Reply:
@ajbeamon Yes, changed the return value to 1.6